version: "3.8"

services:
  db:
    image: postgres:15
    container_name: postgres_db
    restart: always
    environment:
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_DB: ${POSTGRES_DB}
    volumes:
      - db-data:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${POSTGRES_USER} -d ${POSTGRES_DB} -h localhost"]
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
      - backend-net

  backend:
    build:
      context: .
      dockerfile: Dockerfile
    container_name: prestamos-bk
    restart: unless-stopped
    env_file:
      - .env
    environment:
      POSTGRES_HOST: db
      POSTGRES_PORT: 5432
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_DB: ${POSTGRES_DB}
      DATABASE_URL: ${DATABASE_URL}
      PORT: ${PORT}
      NODE_ENV: ${NODE_ENV}
    depends_on:
      db:
        condition: service_healthy
    networks:
      - backend-net

  nginx:
    image: nginx:alpine
    container_name: nginx_reverse
    restart: unless-stopped
    ports:
      - "80:80"
      - "443:443"
    volumes:
      - ./nginx/conf.d:/etc/nginx/conf.d:ro
      - ./nginx/certs:/etc/nginx/certs
      - ./nginx/log:/var/log/nginx
    depends_on:
      - backend
    networks:
      - backend-net

volumes:
  db-data:

networks:
  backend-net:
